sm-core/src/main/java/com/salesmanager/core/business/modules/cms/category/digitalocean/DOSpaceCategoryImageFileManager.java

```diff
@@ -0,0 +1,246 @@
+
+
+      s3.putObject(request);
+
+
+      LOGGER.info("Category add image");
+
+    } catch (final Exception e) {
+      LOGGER.error("Error while adding image to category", e);
+      throw new ServiceException(e);
+
+    }
+  }
+
+  private Bucket getBucket(String bucket_name) {
+    final AmazonS3 s3 = s3Client();
+    Bucket named_bucket = null;
+    List<Bucket> buckets = s3.listBuckets();
+    for (Bucket b : buckets) {
+      if (b.getName().equals(bucket_name)) {
+        named_bucket = b;
+      }
+    }
+
+    if (named_bucket == null) {
+      named_bucket = createBucket(bucket_name);
+    }
+
+    return named_bucket;
+  }
+
+  private Bucket createBucket(String bucket_name) {
+    final AmazonS3 s3 = s3Client();
+    Bucket b = null;
+    if (s3.doesBucketExistV2(bucket_name)) {
+      System.out.format("Bucket %s already exists.\n", bucket_name);
+      b = getBucket(bucket_name);
+    } else {
+      try {
+        b = s3.createBucket(bucket_name);
+      } catch (AmazonS3Exception e) {
+        System.err.println(e.getErrorMessage());
+      }
+    }
+    return b;
+  }
+
+  /**
+   * Builds an amazon S3 client
+   * 
+   * @return
+   */
+  private AmazonS3 s3Client() {
+    AmazonS3 s3 = AmazonS3ClientBuilder
+            .standard()
+            .withEndpointConfiguration(
+                    new AwsClientBuilder.EndpointConfiguration(
+                            this.cmsManager.getBaseUrl(),
+                            regionName()))
+            .withCredentials(new EnvironmentVariableCredentialsProvider())
+            .build();
+
+    return s3;
+  }
+
+  private String bucketName() {
+    String bucketName = getCmsManager().getRootName();
+    if (StringUtils.isBlank(bucketName)) {
+      bucketName = DEFAULT_BUCKET_NAME;
+    }
+    return bucketName;
+  }
+
+  private String regionName() {
+    String regionName = getCmsManager().getLocation();
+    if (StringUtils.isBlank(regionName)) {
+      regionName = DEFAULT_REGION_NAME;
+    }
+    return regionName;
+  }
+
+  private String nodePath(String categoryCode) {
+    return new StringBuilder().append(ROOT_NAME).append(Constants.SLASH).append(categoryCode)
+        .append(Constants.SLASH).toString();
+  }
+
+  private String nodePath(String categoryCode, String categoryImageId) {
+
+    StringBuilder sb = new StringBuilder();
+
+    String nodePath = nodePath(categoryCode);
+    sb.append(nodePath).append(categoryImageId).append(Constants.SLASH);
+
+    return sb.toString();
+
+
+  }
+
+  public static String getName(String filename) {
+    if (filename == null) {
+      return null;
+    }
+    int index = indexOfLastSeparator(filename);
+    return filename.substring(index + 1);
+  }
+
+  public static int indexOfLastSeparator(String filename) {
+    if (filename == null) {
+      return -1;
+    }
+    int lastUnixPos = filename.lastIndexOf(UNIX_SEPARATOR);
+    int lastWindowsPos = filename.lastIndexOf(WINDOWS_SEPARATOR);
```

Comment:
is this meant to be for server, we will never use windows on server.

Reply:
I think these two methods are not used right now, I think there were used before but now no need for theme, we can remove them.